ytgov/yhsi
# Set the host port from an .env file

# User Story
A developer can set the "host port" in a production setting via an `.env` file variable.

# Details
The port can also be set via an exported environment variable. See https://docs.docker.com/compose/environment-variables/
e.g.
```bash
export HOST_PORT=12345
docker-compose -f docker-compose.production.yml up --build -d
```

or for a one off
`HOST_PORT=12345 docker-compose -f docker-compose.production.yml up --build -d`

# Testing Instructions
1. Set the environment variables for the back-end.

```
cp /src/api/.env /src/api/.env.production
vi /src/api/.env.production
```

2. Set the global host port.

```bash
cp .env.sample .env
vi .env
```

> Note that the file needs to be named `.env` and it needs to be at the same level as
> the `docker-compose.production.yml` file.

3. Start the application with:

```
docker-compose -f docker-compose.production.yml up --build -d
```
